src/ParallelAlgorithms/NonlinearSolver/NewtonRaphson/LineSearch.hpp

```diff
@@ -0,0 +1,21 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+namespace NonlinearSolver::newton_raphson {
+/*!
+ * \brief Find the next step length for the line-search globalization
+ *
+ * The step length is chosen such that it minimizes the quadratic (first
+ * globalization step) or cubic (subsequent globalization steps) polynomial
+ * interpolation. This function implements Algorithm A6.1.3 in
+ * \cite DennisSchnabel (p. 325).
+ */
+double line_search(size_t globalization_iteration_id, double step_length,
```

Comment:
Yes, it returns the _next_ step length along the line search. The inputs are the current step length and the previous. The "next" is in the \brief. Also added it in the line following the \brief now. I could also change the function name if you have a suggestion, but the algorithm I'm referring to in the book is called "line search" and I thought that already implies stepping along a line, so returning a step length wouldn't be too unsurprising to return.

Reply:
With the changes you've made I am satisfied.

I think my suggestion would have been to call the function "next_step_length", because that's what is returns, and describe the line search algorithm in the documentation, because that's an implementation detail. Feel free to change it this way if you want to, but I'm not requesting further changes here :)